### PR TITLE
LOC 143 - Create smart contract for hotel reservation, migrations and…

### PR DESCRIPTION
**Pull request by maptuhec**
… unit tests

### CODE REVIEW

File: contracts/HotelReservation/HotelReservationProxy.sol [hunk 1]
```diff
@@ -0,0 +1,20 @@
+pragma solidity ^0.4.17;
+
+import "./../Upgradeability/Forwardable.sol";
+import "./HotelReservationFactory/IHotelReservationFactory.sol";
+
+contract HotelReservationProxy is Forwardable {
+	address public hotelReservationFactoryAddress;
+	IHotelReservationFactory hotelReservationContract;
```
Comment (VladislavIvanov) [inline contracts/HotelReservation/HotelReservationProxy.sol]: Rename to hotelReservationFactoryContract

File: contracts/HotelReservation/HotelReservation.sol [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	uint reservationCostLOC;
+	uint reservationStartDate;
+	uint reservationEndDate;
+	uint daysBeforeStartForRefund;
+	uint refundPercantage;
+	bytes32 hotelId;
+	bytes32 roomId;
+	address hotelReservationFactoryAddress;
+
+	event LogCreateHotelReservation(bytes32 _hotelId, bytes32 _roomId);
```
Comment (thcrnk) [inline contracts/HotelReservation/HotelReservation.sol]: Change event to be as it is in HotelReservationFactory - with the new fields

File: contracts/HotelReservation/IHotelReservation.sol [hunk 3]
```diff
@@ -0,0 +1,20 @@
+pragma solidity ^0.4.17;
+
+import "./../Upgradeability/OwnableUpgradeableImplementation/IOwnableUpgradeableImplementation.sol";
+
+contract IHotelReservation is IOwnableUpgradeableImplementation {
+
+event LogCreateHotelReservation(bytes32 _hotelId, bytes32 _roomId);
```
Comment (thcrnk) [inline contracts/HotelReservation/IHotelReservation.sol]: Change event

File: test/HotelReservationFactory.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+	let hotelReservationFactory;
+	let hotelReservationFactoryProxy;
+	let hotelReservationFactoryImpl;
+
+
+	const _owner = accounts[0];
+	const _notOwner = accounts[1];
+
+	describe("create hotel room factory proxy", () => {
```
Comment (thcrnk) [inline test/HotelReservationFactory.js]: Change name of describe to be for reservation

File: test/HotelReservationFactory.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+			await hotelReservationContract.init();
+			await hotelReservationContract.setImplAddress(hotelReservation.address);
+		});
+
+		it("should get the owner of the first contract", async function () {
+			const owner = await hotelReservationContract.getOwner();
+			assert.strictEqual(owner, _owner, "The owner is not set correctly");
+		});
+	});
+
```
Comment (thcrnk) [inline test/HotelReservationFactory.js]: Add tests to upgrade the contract - upgrade, throw when non owner is upgrading and etc.

File: contracts/HotelReservation/HotelReservationFactory/HotelReservationFactory.sol [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+		bytes32 _hotelReservationId,
+		address _customerAddress,
+		uint _reservationCostLOC,
+		uint _reservationStartDate,
+		uint _reservationEndDate,
+		uint _daysBeforeStartForRefund,
+		uint _refundPercantage,
+		bytes32 _hotelId,
+		bytes32 _roomId
+	) public returns(bool success)
```
Comment (thcrnk) [inline contracts/HotelReservation/HotelReservationFactory/HotelReservationFactory.sol]: Add modifier to check if this _hotelReservationId already exists, should be new id

File: test/HotelReservation.js [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+				reservationCostLOC,
+				reservationStartDate,
+				pastDate,
+				daysBeforeStartForRefund,
+				refundPercantage,
+				hotelId,
+				roomId
+			));
+		})
+	})
```
Comment (thcrnk) [inline test/HotelReservation.js]: Test upgradability as it is in Rental tests

**Pull request merged**